### DISCUSSION TITLE
The time step of ITI Barge

### DISCUSSION THREAD

**Discussion by icnosvbovwb** [OpenFAST/openfast#1628] [Q&A]
Dear @jjonkman 
I calculated the ITI Barge wind turbine by modifying DT=0.05s, but the result showed an error in the image. The default time step was 0.025s, and there was no problem calculating with the time step of 0.025s. Can you tell me the reason? Or how to increase the time step to 0.05 seconds so that it can be calculated. Or does it have a maximum time step of 0.025 seconds？
![da3b3187eeee2010303aa0a0fb7904b](https://github.com/OpenFAST/openfast/assets/107483730/c25babd5-1f72-4c82-84cf-ec090e12eabd)

**Comment by jjonkman** [comment]
Dear @icnosvbovwb,

Actually, for the OpenFAST model of the NREL 5-MW baseline wind turbine atop the ITI Energy barge in the r-test (https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/5MW_ITIBarge_DLL_WTurb_WavesIrr), the default time step is 0.0125 s (not 0.025 s).  As has been discussed on our forum, the required time step is intrinsically tied to the natural frequencies inherent in the model: https://forums.nrel.gov/t/fast-integrator-step-size-precision/247.  For the ITI Energy barge model, I would guess the nacelle-yaw degree of freedom is driving the highest natural frequency, so, e.g.,, setting `YawDOF` = FALSE may allow you to increase the time step in this model.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
I still encountered the same problem by setting YawDOF = FALSE. Can you tell me how to achieve a time step of 0.05s?
or is it impossible to achieve a higher time step?
![image](https://github.com/OpenFAST/openfast/assets/107483730/50c67397-ae5e-415a-bf89-d82ad9605a5a)
![image](https://github.com/OpenFAST/openfast/assets/107483730/b1a4000c-d27d-41e8-87f3-1d3c349c6b9e)
![image](https://github.com/OpenFAST/openfast/assets/107483730/ef6e888e-6138-4a95-bd49-388ad92c7ef3)

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
![image](https://github.com/OpenFAST/openfast/assets/107483730/8f7ca963-0ead-42b8-8495-f1884d64c895)
![image](https://github.com/OpenFAST/openfast/assets/107483730/0849c62f-62bc-4650-9e0b-536bb6a405c8)
![image](https://github.com/OpenFAST/openfast/assets/107483730/fe919651-4ac7-4c7c-8206-29991a92328a)
When I set the switch as shown in the figure, the wind turbine can be calculated according to the time steps of 0.05s(FlapDOF1 = FlapDOF2 = EdgeDOF = TeetDOF = TwFADOF2 = TwSSDOF2 =False ), and the result is almost consistent with the calculation of 0.025s(Only TeetDOF = False ). I am not sure why.

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

I didn't say that setting `YawDOF` = FALSE will allow you to increase the time step to 0.05 s; I only said that it may allow you to increase the time step above 0.0125 s.  You'd have to figure out the highest remaining natural frequency (e.g., by post-processing a linearization analysis) to identify the highest natural frequency you can increase the time step to.

It looks like disabling the blade and 2nd tower-bending DOFs allows you to increase the time step as you want.  It makes sense to me that disabling the 2nd blade flapwise and 2nd tower-bending DOFs would enable the use of a large time step, but I'd be surprised if the 1st blade flapwise and edgewise DOFs would need to be disabled, at least for the NREL 5-MW baseline wind turbine model.

Regarding the time series you show, it could be that these responses are not sensitive to the DOFs you've disabled for this specific simulation.  To properly assess the relative importance of various DOFs, I would suggest looking at motions and loads throughout the system for a variety of load cases.

Best regards
